File: project-service/src/main/java/com/kapitonau/projectservice/repository/ProjectRepository.java
package com.kapitonau.projectservice.repository;

import com.kapitonau.projectservice.model.ProjectModel;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface ProjectRepository extends JpaRepository<ProjectModel, Long> {

    Optional<ProjectModel> findByTitleAndSpaceId(String title, Long spaceId);

}

File: workspace-service/src/main/kotlin/com/kapitonau/ps/workspaceservice/repository/ProjectRepository.kt
package com.kapitonau.ps.workspaceservice.repository

import com.kapitonau.ps.workspaceservice.model.ProjectModel
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface ProjectRepository : JpaRepository<ProjectModel, Long> {

    fun findByProjectTitle(title: String): Optional<ProjectModel>

}
